documentation/private-link-service/private-link-redirect-map.md

```diff
@@ -0,0 +1,261 @@
+
+![PL Redirect VM-to-PLS direction](images/private-link-redirect-vm-to-pls.svg)
+
+On the reverse direction, it reverses all the actions in the forwarding side, e.g., 6to4, as it shows below:
+
+![PL Redirect PLS-to-VM direction](./images/private-link-redirect-pls-to-vm.svg)
+
+### 3.2 Redirect map with PL NSG
+
+When PL NSG is enabled, the extra encap for tunneling the packet to NSG will still be added on top the original PL encap. And the return packet will be exactly the same as the regular case without PL NSG.
+
+### 3.3 Redirect map with fast path
+
+Please refer to [private link load balancer fast path support](./private-link-service.md#33-load-balancer-fast-path-support).
+
+### 3.4 Non-required features
+
+- RST on connection idle timeout.
+- Flow resimulation should not be triggered automatically on redirect map update.
+
+## 4. Resource modeling, requirement, and SLA
+
+### 4.1 Scaling requirement
+
+For scaling requirement, please refer to [SONIC DASH HLD](https://github.com/sonic-net/DASH/blob/main/documentation/general/dash-sonic-hld.md#14-scaling-requirements).
+
+### 4.2 Reliability requirements
+
+The flows created when redirect map will be replicated among multiple DPUs, following the SmartSwitch HA design.
+
+For more information, please refer to [SmartSwitch HA design doc](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/high-availability/smart-switch-ha-hld.md).
+
+## 5. SAI API design
+
+### 5.1. DASH Outbound Port Map
+
+A new object `outbound-port-map` is added to manage port mappings in the outbound direction, which includes the following attributes:
+
+| Attribute name | Type | Description |
+| --- | --- | --- |
+| SAI_OUTBOUND_PORT_MAP_ATTR_COUNTER_ID | sai_object_id_t | A counter that tracks the number of packets and bytes associated with the matching outbound-port-map. |
+
+### 5.2. DASH Outbound Port Map Port Range
+
+A new object key entry `outbound-port-map-port-range` is designed to define port range translation in the outbound direction. It is particularly useful for scenarios involving Private Link (PL) or Private Link Network Security Group (PL NSG) redirect maps. The key entry for this object specifies how port ranges should be translated when traffic is directed outbound.
+
+The entry is defined like below:
+
+```c
+typedef struct _sai_outbound_port_map_port_range_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Exact matched key outbound_port_map_id
+     *
+     * @objects SAI_OBJECT_TYPE_OUTBOUND_PORT_MAP
+     */
+    sai_object_id_t outbound_port_map_id;
+
+    /**
+     * @brief Range matched key dst_port_range
+     */
+    sai_u32_range_t dst_port_range;
```

Comment:
Is the port range inclusive or exclusive?

Reply:
inclusive